decorators_library/decorators.py

```diff
@@ -1 +1,71 @@
 # implement your decorators here.
+#import signal
+import time
+import logging
+from exceptions import TimeoutError
+from threading import Thread
+
+# def timeout(seconds): # PROBLEMS: won't raise error
+#     def wrapper(func):
+#         def _new_func(*args, **kwargs):
+#             try:
+#                 func(*args, **kwargs)
+#             except TimeoutError:
+#                 raise 'Function call timed out'
+#         timer = Thread(target=_new_func)
+#         timer.daemon = True
+#         try:
+#             timer.start()
+#             timer.join(seconds)
+#         except TimeoutError:
+#             raise TimeoutError('Function call timed out')
+#         return _new_func
+#     try:
+#         return wrapper
+#     except TimeoutError:
+#         raise TimeoutError('Function call timed out')
+        
+def debug():
+    def finished_func(func):
+        def _new_func(*args, **kwargs):
+            logging.debug('Executing {} with {}, {}'.format(func, *args, **kwargs))
+            return func(*args, **kwargs)
```

Comment:
How does this work? Does passing func in line 31 end up calling func.__name__ or something to create a string?